### PR TITLE
Some very useful functions

### PR DESCRIPTION
**Pull request by andersonfaaria**
- New isWalkable function that replaces several other constructed ones and has all the major checks
- getDirectionTo which returns the direction you need to face to start moving from pos1 to pos2
- pairsByKeys which allows you to iterate through tables using keys returned by a given sort function
- Plus some fixes in identation of resonance chamber

### CODE REVIEW

File: data/lib/core/tile.lua [hunk 1]
```diff
@@ -36,18 +36,24 @@ function Tile.relocateTo(self, toPosition)
...
+		if topStackItem and topStackItem:hasProperty(CONST_PROP_BLOCKPATH) then return false end
 	end
-
-	local items = self:getItems()
-	for i = 1, self:getItemCount() do
-		local item = items[i]
-		local itemType = item:getType()
-		if itemType:getType() ~= ITEM_TYPE_MAGICFIELD and not itemType:isMovable() and item:hasProperty(CONST_PROP_BLOCKSOLID) then
-			return false
+	if proj then
```
Comment (jlcvp) [inline data/lib/core/tile.lua:50]: what does `proj` stands for?

Comment (andersonfaaria) [reply data/lib/core/tile.lua:50]: projectile, if set as true it means we will check for blocksolid property in items (besides the usual check in the tile itself at the beggining)

File: data/lib/core/position.lua [hunk 2]
```diff
@@ -127,14 +83,20 @@ function Position:moveDownstairs()
...
+	
+    if not tile:isWalkable(false, false, false, false, true) then
         for direction = DIRECTION_NORTH, DIRECTION_NORTHEAST do
             if direction == DIRECTION_SOUTH then
                 direction = DIRECTION_WEST
             end
 
             local position = self + Position.directionOffset[direction]
-            if isWalkable(position) then
+			local new_tile = Tile(position)
```
Comment (dudantas) [inline data/lib/core/position.lua]: Wouldn't it be interesting to switch to newTile?

**Review by dudantas — APPROVED** [review]
For my tests it's ok.
There is only the suggestion to change the new_tile variable to newTile

**Pull request merged**